### PR TITLE
Make the `forkTxProducer` in ThreadNet tests get a `BackingStoreValueHandle` on each slot

### PR DESCRIPTION
**Pull request by Jasagredo**
# Description

Retrieve a BackingStoreValueHandle on each act of the watcher in `forkTxProducer` in the ThreadNet tests

### CODE REVIEW

File: ouroboros-consensus-test/src/Test/ThreadNet/Network.hs [hunk 1]
```diff
@@ -1044,29 +1044,10 @@ runThreadNetwork systemTime ThreadNetworkArgs
...
-             )
-        extractBsValueHandle = \case
-            StaticLeft bsValueHandle -> bsValueHandle
-      seBsValueHandle <-
-          fmap extractBsValueHandle
-        $ ChainDB.getLedgerBackingStoreValueHandle chainDB registry (StaticLeft ())
+      let getValueHandle = do
+            bsvh <- ChainDB.getLedgerBackingStoreValueHandle chainDB registry (StaticLeft ())
+            pure $ mkDiskLedgerView $ case bsvh of
+              StaticLeft v -> v
```
Comment (jorisdral) [inline ouroboros-consensus-test/src/Test/ThreadNet/Network.hs:1050]: What do you think of adding these functions to `Ouroboros.Consensus.Util`:

```haskell
staticFromLeft :: StaticEither 'False l r -> l
staticFromLeft = \case (StaticLeft x) -> x

staticFromRight :: StaticEither 'True l r -> r
staticFromRight = \case (StaticRight x) -> x
```

**Review by jorisdral — APPROVED** [review]
Could you elaborate on why this change is necessary? (not that I'm doubting it, but I don't know the motivation)

File: ouroboros-consensus-test/src/Test/ThreadNet/Network.hs [hunk 2]
```diff
@@ -658,13 +658,13 @@ runThreadNetwork systemTime ThreadNetworkArgs
...
                    -> Seed
-                   -> DiskLedgerView m (ExtLedgerState blk)
+                   -> m (DiskLedgerView m (ExtLedgerState blk))
                       -- ^ How to get the current ledger state
                    -> Mempool m blk
                    -> m ()
     forkTxProducer coreNodeId registry clock cfg nodeSeed dlv mempool =
         void $ OracularClock.forkEachSlot registry clock "txProducer" $ \curSlotNo -> do
-          let (DiskLedgerView emptySt _ doRangeQuery _) = dlv
+          DiskLedgerView emptySt _ doRangeQuery _ <- dlv
```
Comment (dnadales) [inline ouroboros-consensus-test/src/Test/ThreadNet/Network.hs:667]: Minor: would it make sense to replace `dlv` with `mdlv`?

File: ouroboros-consensus-test/src/Test/ThreadNet/Network.hs (hunk 1 above)
Comment (dnadales) [reply ouroboros-consensus-test/src/Test/ThreadNet/Network.hs:1050]: I like the idea.

**Review by dnadales — APPROVED** [review]
Thanks!

**Pull request merged**